src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+var setval = require('setval')
+var yo = require('yo-yo')
+var csjs = require('csjs-inject')
+var Treeview = require('ethereum-remix').ui.TreeView
+
+var css = csjs`
+  .treeview {
+    display: block;
+    margin-top: 7px;
+    width: 20%;
+  }
+  .folder,
+  .file {
+    font-size: 14px;
+  }
+  .hasFocus {
+    background-color: #F4F6FF;
+  }
+  .rename {
+    background-color: white;
+  }
+  .remove,
+  .save {
+    align-self: center;
+    margin-right: 3%;
+  }
+  .activeMode {
+    display: flex;
+    justify-content: space-between;
+  }
+`
+module.exports = fileExplorer
+
+function fileExplorer (container, files, editor) {
+
+  container.className = css.treeview
+
+  var sources = {
+    'test/client/credit.sol': '',
+    'src/voting.sol': '',
+    'src/leasing.sol': '',
+    'src/gmbh/contract.sol': false,
+    'src/gmbh/test.sol': false,
+    'src/gmbh/company.sol': false,
+    'src/gmbh/node_modules/ballot.sol': false,
+    'src/ug/finance.sol': false,
+    'app/solidity/mode.sol': true,
+    'app/ethereum/constitution.sol': true
+  }
+  Object.keys(sources).forEach(function (key) { files.set(key, sources[key]) })
+
+  var deleteButton = yo`<span class=${css.remove} onclick=${deletePath}><i class="fa fa-trash" aria-hidden="true"></i></span>`
+
+  // files.event.register('fileChanged', function (filepath) { })
+  files.event.register('fileAdded', function (filepath) {
+    console.log('fileAdded', filepath)
+  })
+  files.event.register('fileFocus', function (path) {
+    if (filepath == path) return
+    filepath = path
+    var el = getElement(filepath)
+    var tmp = el
+    while ((tmp = tmp.parentElement.parentElement) && tmp.tagName === 'LI') {
+      var caret = tmp.firstChild.firstChild
+      if (caret.classList.contains('fa-caret-right')) caret.click() // expand
+    }
+    el.click() // focus element
+  })
+  files.event.register('fileRemoved', function (filepath) {
+    console.log('fileRemoved', filepath)
+  })
+  files.event.register('fileRenamed', function (oldName, newName) {
+    var el = getElement(oldName)
+    if (el) {
+      oldName = oldName.split('/')
+      newName = newName.split('/')
+      var index = oldName.reduce(function (idx, key, i) {
+        return oldName[i] !== newName[i] ? i : idx
+      }, undefined)
+      var newKey  = newName[index]
+      var oldPath = oldName.slice(0,index+1).join('/')
```

Comment:
Please have spacing within parameters and arithmetics.

All this should validate with `standard`, I thought it would disallow `(0,index+1)`.